src/prefect/core/flow.py

```diff
@@ -824,30 +854,30 @@ def _sorted_tasks(self, root_tasks: Tuple[Task, ...] = None) -> Tuple[Task, ...]
 
     def set_dependencies(
         self,
-        task: object,
-        upstream_tasks: Iterable[object] = None,
-        downstream_tasks: Iterable[object] = None,
-        keyword_tasks: Mapping[str, object] = None,
+        task: Any,
+        upstream_tasks: Iterable[Any] = None,
+        downstream_tasks: Iterable[Any] = None,
```

Comment:
unrelated to this PR but since you've changed it: shouldn't these be `Task` and `Iterable[Task]`? What am I missing?

Reply:
`as_task()` is called on these inputs, both in this function (for the main "task") and in the Edge constructor, so this function can accept any object.